processing/src/main/java/org/apache/druid/query/aggregation/first/StringFirstLastUtils.java

```diff
@@ -59,6 +61,46 @@ public static boolean selectorNeedsFoldCheck(
            || SerializablePairLongString.class.isAssignableFrom(clazz);
   }
 
+  /**
+   * Returns whether an object *might* contain SerializablePairLongString objects.
+   */
+  public static boolean objectNeedsFoldCheck(Object obj)
+  {
+    if (obj == null) {
+      return false;
+    }
+    final Class<?> clazz = obj.getClass();
+    return clazz.isAssignableFrom(SerializablePairLongString.class)
+           || SerializablePairLongString.class.isAssignableFrom(clazz);
+  }
+
+  /**
+   * Return the object at a particular index from the vector selectors
+   */
```

Comment:
Important comment for this javadoc - index of bounds issues is the responsibility of the caller

Reply:
Updated